File: 1518. Water Bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int):
        final_ans = numBottles

        while(numBottles>=numExchange):
            q=numBottles//numExchange
            r=numBottles%numExchange
            final_ans+=q
            numBottles=q+r
        return(final_ans)

File: 1518. Water Bottles.java
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        drinked = 0

        empty = 0
        full = numBottles 

        while full > 0:
            drinked += full
            empty += full

            N = empty // numExchange
            full = N
            empty -= N * numExchange
        
        return drinked
